### PR TITLE
Do not require new headers if lane is empty

### LINKED ISSUE #1723: Parachain heads are synced without demand in parachain <> parachain bridge
It seems that when we bridge two parachains and run a complex relayer, parachains heads are relayed even when this isn't required by the message relay. It needs to be investigated and fixed.

### PR DESCRIPTION
**Pull request by svyatonik**
closes #1723

If latest confirmed nonce is `0`, it means that nothing has been actually confirmed. So we do not need to ask for new headers just because we have nothing to deliver

### CODE REVIEW

**Review by acatangiu — APPROVED** [review]

**Comment by svyatonik** [comment]
Sorry, the fix is not complete (on a second thought). Will come back with proper fix

**Comment by svyatonik** [comment]
Now it should be fine - I've remembered the similar #1328, so the fix should be general. But let's not merge this PR until next week - if something will break, I want to be sure that it is the #1669, not this PR

**Pull request merged**
